packages/react/src/components/teleport/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { createPortal } from "react-dom"
+import { findOrCreateTeleportTarget } from "./util"
+import { ReactNode } from "react"
+
+export interface TeleportProps {
+  to: string
+  children: ReactNode
+  renderKey?: string | null | undefined
+}
+
+/**
+ * Renders child content in a portal element at the DOM root
+ * 
+ * @param to The destination portal name
+ * @param children The content to be teleported
+ * @param renderKey (optional) A key to control when the content rerenders
+ */
+export const Teleport = ({ to, children, renderKey }: TeleportProps) => {
```

Comment:
Following @ahmed-gz's comments - we should make it clear that `to` is an element ID.

Reply:
Whats the benefit of this if `<Teleport>` creates the portal element for you?

I don't think I understand why it matters what the selector scheme is

Reply:
After taking a look at how you used this component, I think I have a better understanding of the wanted level of abstraction, I was looking at it on a lower level. `to` scheme doesn't really matter here since that's only a name used internally to identify the container. All good, thank you!

Reply:
Sorry if that reply came off defensive.

I had a think about this while I was out and a benefit would be that we allow `<Teleport>` to be used with existing elements which weren't created using `<Teleport>`. In theory this kind of API should probably also support passing an element directly.

I'm leaning towards us not supporting this feature though, you could just use `createPortal()` for both of these use cases.

Thought dump:
I see `<Teleport>` as a "managed portal" in concept, so we want to retain control over the portal generation and resolution. For example in future we might change the id generation scheme i.e. `slashid-teleport-${to}` to reduce collisions, or decorate the portal with some css etc.

Reply:
Okay all that said I will leave it as is for now, lets revisit this with hindsight once it's in use

Reply:
No worries, all good! I wrote the comment before I fully understood the idea behind it - once I saw how you used it in context of `<Dialog>` it clicked with me. Initially I assumed the container would be created externally and this would just lock on it, but it makes perfect sense to consider this as a managed component. I think we should leave it as it is, but maybe add a note in the TS docs for clarity?

Reply:
Good idea, done!